cryptol-remote-api/python/cryptol/cryptoltypes.py

```diff
@@ -105,22 +117,19 @@ def __to_cryptol_str__(self) -> str:
         else:
             return ' '.join(parenthesize(x.__to_cryptol_str__()) for x in [self._rator, *self._rands])
 
-    def __eq__(self, other : Any) -> bool:
-        return isinstance(other, CryptolApplication) and self._rator == other._rator and self._rands == other._rands
-
-    def __repr__(self) -> str:
-        return f'CryptolApplication({", ".join(repr(x) for x in [self._rator, *self._rands])})'
 
+# -----------------------------------------------------------
+# Cryptol kinds
+# -----------------------------------------------------------
 
+@dataclass
 class CryptolArrowKind:
-    def __init__(self, dom : CryptolKind, ran : CryptolKind):
-        self.domain = dom
-        self.range = ran
+    domain : CryptolKind
+    range : CryptolKind
 
-    def __repr__(self) -> str:
-        return f"CryptolArrowKind({self.domain!r}, {self.range!r})"
-
-CryptolKind = Union[Literal['Type'], Literal['Num'], Literal['Prop'], CryptolArrowKind]
+CryptolKind = Union[typing_extensions.Literal['Type'],
+                    typing_extensions.Literal['Num'],
+                    typing_extensions.Literal['Prop'], CryptolArrowKind]
```

Comment:
can `CryptolArrowKind` live with it's union-mates on it's own new line too? minor I know xD sorry